utils/src/main/java/com/cloud/utils/ssh/SSHCmdHelper.java

```diff
@@ -77,8 +80,47 @@ public String getStdErr() {
     }
 
     public static com.trilead.ssh2.Connection acquireAuthorizedConnection(String ip, int port, String username, String password) {
+        return acquireAuthorizedConnection(ip, 22, username, password, null);
+    }
+
+    public static boolean acquireAuthorizedConnectionWithPublicKey(final com.trilead.ssh2.Connection sshConnection, final String username, final String privateKey) {
+        if (StringUtils.isNotBlank(privateKey)) {
+            File privateKeyFile = null;
+            try {
+                privateKeyFile = File.createTempFile("cloudstack-host-", null);
```

Comment:
@weizhouapache 
This temp. hack may not be required as mgmt server saves the ssh private/pub keys at: `/var/lib/cloudstack/management/.ssh/id_rsa` on the mgmt server, why not just read/use that?

In fact if this is saved/assumed, then we don't even need to read the private key from db and pass it here. Thoughts?

Reply:
Note: needs checking on all distros (this path), on centos7, centos/rocky8, ubuntu and suse. cc @davidjumani if you know/remember distro specific paths.

Reply:
> @weizhouapache This temp. hack may not be required as mgmt server saves the ssh private/pub keys at: `/var/lib/cloudstack/management/.ssh/id_rsa` on the mgmt server, why not just read/use that?
> 
> In fact if this is saved/assumed, then we don't even need to read the private key from db and pass it here. Thoughts?

@rhtyd 
I have considered it. but at the end I decided to use temp file so that the process will not rely on any file on management server.

Reply:
In that case let's fix the mod bits of the file to be just read-only (0400)